cmd/minishift/cmd/root.go

```diff
@@ -213,14 +222,38 @@ func initializeProfile() string {
 		activeProfile string
 	)
 
+	// If `minishift profile set` or `minishift start --profile` command is used.
 	for i, arg := range os.Args {
+		// This will match if `--profile` flag is used
 		if arg == "--"+profileFlag {
+			// This make sure if user specify `--profile` flag without any name
+			// it should not panic with out of index error
+			if len(os.Args) <= i+1 {
+				break
+			}
 			profileName = os.Args[i+1]
 			if !cmdUtil.IsValidProfileName(profileName) {
 				atexit.ExitWithMessage(1, "Profile names must consist of alphanumeric characters only.")
 			}
 			break
 		}
+		// This will match if we used profile command
+		if arg == profileFlag || arg == profileAlias[0] || arg == profileAlias[1] {
+			// This make sure if user specify profile command without any subcommand then
+			// it should not panic with out of index error
+			if len(os.Args) <= i+2 {
+				break
+			}
+			// This will match if after profile command we used set subcommand
+			// Then it will have profileName as whatever we provided as set command parameter.
```

Comment:
This comment is not easy to understand. Suggestion, For use cases when `--profile PROFILE_NAME` is used with `minishift profile set` command.